Taarafo.Core.Tests.Unit/Services/Foundations/Comments/CommentServiceTests.Validation.Modify.cs

```diff
@@ -369,5 +370,67 @@ public async Task ShouldThrowValidationExceptionOnModifyIfStorageUpdatedDateSame
             this.dateTimeBrokerMock.VerifyNoOtherCalls();
             this.loggingBrokerMock.VerifyNoOtherCalls();
         }
+
+        [Fact]
+        public async void ShouldThrowValidationExceptionOnModifyWhenReferneceExceptionAndLogItAsync()
+        {
+            // given
+            int minuteInPast = GetRandomNegativeNumber();
+            DateTimeOffset randomDate = GetRandomDateTimeOffset();
+            Comment randomComment = CreateRandomModifyComment(randomDate.AddMinutes(minuteInPast));
+            Comment inputComment = randomComment.DeepClone();
+            inputComment.UpdatedDate = randomDate;
+            Comment storageComment = randomComment;
+            string randomMessage = GetRandomMessage();
+            string exceptionMessage = randomMessage;
+            var foreignKeyConstraintConflictException = new ForeignKeyConstraintConflictException(exceptionMessage);
+
+            var invalidCommentReferenceException =
+                new InvalidCommentReferenceException(foreignKeyConstraintConflictException);
+
+            var expectedCommentValidationException =
+                new CommentValidationException(invalidCommentReferenceException);
+
+            this.dateTimeBrokerMock.Setup(broker =>
+                broker.GetCurrentDateTimeOffset())
+                    .Returns(randomDate);
```

Comment:
We shall throw the exception in the first dependency call.

Reply:
@ShriHumrudha I will update,  can you explain why we throw this there?   I will also have to update this on the other one as that is also done on the storageBroker rather than the 1st one.